### ISSUE TITLE
GetProcAddress failed

### ISSUE THREAD

**Issue by antoniu200** [VaanaCZ/LANVP#24]
Hi,

I am using Windows 10 64 bit 1607, L.A. Noire 2675, LANVP v1.1b. Getting this message twice before the game launches: one right before the Rockstar Launcher window appears on screen, then once again right before the game window launches.

EDIT: Forgot to mention, using keyboard and mouse.

---------------------------
GetProcAddress failed!
---------------------------
GetdfDIJoystick
---------------------------
OK   
---------------------------

**Comment by 18iteration** [comment]
> Hi,
> 
> I am using Windows 10 64 bit 1607, L.A. Noire 2675, LANVP v1.1b. Getting this message twice before the game launches: one right before the Rockstar Launcher window appears on screen, then once again right before the game window launches.
> 
> EDIT: Forgot to mention, using keyboard and mouse.
> 
> ## GetProcAddress failed!
> ## GetdfDIJoystick
> ## OK

May you provide the full error or even better, log file?
(I don't remember where the logs are, you'd have to look into the game folder, documents->mygames?rockstar? or %appdata%

**Comment by antoniu200** [comment]
Unfortunately, there are no logs, since the game does not crash. Just the plugin sends this message twice before the game window opens.

**Comment by 18iteration** [comment]
> Unfortunately, there are no logs, since the game does not crash. Just the plugin sends this message twice before the game window opens.

The Windows32 API function `GetProcAddress()` could fail to load, say a dll, for a variety of reasons, but it's nothing to be concerned about, unless you encountered actual issues. Since the game loads up fine and you don't have an issue, I would suggest closing this thread with the _Closing issue_ button.

Have a great day.

**Comment by VaanaCZ** [comment]
For the next releases, I will switch to Ultimate ASI loader, so issues like this should go away.

**Comment by antoniu200** [comment]
Thank you!
Will test, let me know when you publish the next release.
If it's fixed, I will close the issue.

**Comment by VaanaCZ** [comment]
@antoniu200 If you are interested, you can try out a testing version of the next release: https://github.com/VaanaCZ/LANVP/releases/tag/v2.0-RC2

**Comment by antoniu200** [comment]
Sorry for the late reply, I haven't played the game since I opened this issue.

After updating the patch, I get this message every time I launch the game, and the patch version does **not** appear in the lower left corner:

---------------------------
[V-PATCH] Error while initializing patch!
---------------------------
Failed to apply the following patches:

    Show the version number in the main menu
---------------------------
OK   
---------------------------

Also, the borderless patch does not work properly: it ignores me disabling it in `lanvp.ini` and the `Fullscreen` option in the `settings.ini` file gets reset to `NO` no matter what I do, so the game always launches in a window that I can resize to fullscreen by Alt+Enter.

In simpler terms: I want the game to run in Fullscreen mode, but the game / patch does not allow me to.

Anyway, the GetProcAddres failed message seems to have disappeared on version 1.1b after updating to Windows 11 22H2.1485 (I can't imagine how I could have fixed this otherwise, through just file permissions), so I am closing this issue.

The game now works as I would like it to, but, unfortunately, not on the new 2.0 patch.

**Comment by VaanaCZ** [comment]
> Also, the borderless patch does not work properly: it ignores me disabling it in `lanvp.ini` and the `Fullscreen` option in the `settings.ini` file gets reset to `NO` no matter what I do, so the game always launches in a window that I can resize to fullscreen by Alt+Enter.

Make sure you are editing the correct ini. It is now in the plugins folder and not the root directory.

> ## [V-PATCH] Error while initializing patch!
> Failed to apply the following patches:
> 
> ```
> Show the version number in the main menu
> ```

Which version do you have? Are you playing on Steam or R* Launcher?

**Comment by antoniu200** [comment]
> Make sure you are editing the correct ini. It is now in the plugins folder and not the root directory.

Yeah, I know. That's the one I was editing. Also the `settings.ini` inside `Documents\Rockstar Games\L.A. Noire`.

> Which version do you have? Are you playing on Steam or R* Launcher?

2675.1 on Rockstar Launcher (latest, afaik)

**Comment by VaanaCZ** [comment]
Ok, I purchased a second copy of the game on the Rockstar Launcher just to check. There were issues with the implementation on this version but those have been fixed. Here is a new release:

https://github.com/VaanaCZ/LANVP/releases/tag/v2.0-RC3

@antoniu200 Regarding the fullscreen issue, that does not appear to be an issue with the patch, but instead a problem with the launcher.
You have to disable fullscreen directly in the launcher, since it overrides Settings.ini every time the game starts (Settings > My Installed Games > L.A. Noire > Fullscreen)
![image](https://github.com/user-attachments/assets/80b85ec4-d3c6-428e-880a-7b06e94573d9)

I have added a comment about this into the config as a precaution.

**Comment by antoniu200** [comment]
> @antoniu200 Regarding the fullscreen issue, that does not appear to be an issue with the patch, but instead a problem with the launcher. You have to disable fullscreen directly in the launcher, since it overrides Settings.ini every time the game starts (Settings > My Installed Games > L.A. Noire > Fullscreen) ![image](https://private-user-images.githubusercontent.com/14859721/352783935-80b85ec4-d3c6-428e-880a-7b06e94573d9.png)

Interesting, I wasn't aware of this. Thanks for mentioning!

I just tested the RC3, both issues I had above are gone: VPatch version now appears in the bottom left corner. I tested The Set Up and everything worked nicely.

Many thanks for your work!
